service/history/api/multioperation/api.go

```diff
@@ -272,17 +288,13 @@ func startAndUpdateWorkflow(
 
 	switch startOutcome {
 	case startworkflow.NoStart:
+		// An error would have been returned above in this case.
 		panic("unreachable")
 	case startworkflow.StartNew:
-	case startworkflow.StartReused:
-		// The workflow was meant to be *started* - but was actually *not* started since it's already running.
-		// The best way forward is to exit and retry from the top.
-		// By returning an Unavailable service error, the entire MultiOperation will be retried.
-		return nil, startOutcome, newMultiOpError(
-			serviceerror.NewUnavailable("Workflow could not be started as it is already running"), multiOpAbortedErr)
-	case startworkflow.StartDeduped:
-		// Since the start request was deduped, the update was not applied to the *current* workflow execution.
-		// Returning here to allow the caller to apply the update to the current workflow execution.
+		// The workflow was started, as expected.
+	case startworkflow.StartDeduped, startworkflow.StartReused:
+		// The workflow was not started.
+		// Aborting since the update has not been applied.
```

Comment:
Can you return a different error type here for each undesirable outcome from Invoke? Are these outcomes non-error conditions anywhere else?

Reply:
I can introduce a sentinel error if you think that's clearer. So I can distinguish between a "real" error.

Reply:
In my opinion that would be cleaner. You can ignore or fail on these errors.
Let me approve this PR. If you think that is a good idea, you can follow up with a different PR.

Reply:
Done!